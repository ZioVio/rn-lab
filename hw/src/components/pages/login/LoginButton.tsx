import React from 'react';
import { GestureResponderEvent, StyleSheet, Text, View } from 'react-native';
import { TouchableHighlight } from 'react-native-gesture-handler';
import colors from '../../../constants/colors';

interface LoginButtonProps {
  title: string;
  onPress: (ev: GestureResponderEvent) => void;
}

export default function LoginButton({ onPress, title }: LoginButtonProps) {
  const uppercasedTitle = title.toUpperCase();
  return (
    <TouchableHighlight onPress={onPress} testID="LoginButton.TouchableHighlight">
      <View style={styles.button}>
        <Text style={styles.text} testID="LoginButton.Text">{uppercasedTitle}</Text>
      </View>
    </TouchableHighlight>
  );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.colorOrange,
    alignItems: 'center',
    borderRadius: 20,
  },
  text: {
    color: colors.colorPrimary,
    paddingVertical: 2,
    fontSize: 30,
  },
});
