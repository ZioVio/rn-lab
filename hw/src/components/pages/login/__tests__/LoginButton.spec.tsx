import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import LoginButton from '../LoginButton';

test('should match the snapshot', () => {
  const { toJSON } = render(<LoginButton onPress={() => { }} title="some title" />);
  expect(toJSON()).toMatchSnapshot();
});

test('Should call onPress callback', () => {
  const onPress = jest.fn();
  const onPressArg = 'some arg';
  const { getByTestId } = render(<LoginButton onPress={onPress} title="anyTitle"/>);
  fireEvent(getByTestId('LoginButton.TouchableHighlight'), 'press', onPressArg);
  expect(onPress).toHaveBeenCalled();
  expect(onPress).toHaveBeenCalledWith(onPressArg);
});

test('Should have uppercased title', () => {
  const sampleTitles = [
    'title', 'TITLE', 'tIt Le', 'SoMeThInG VeIrD', '',
  ];
  sampleTitles.forEach((title) => {
    const { getByTestId } = render(<LoginButton onPress={() => {}} title={title}/>);
    const buttonTitle = getByTestId('LoginButton.Text')?.children[0]?.toString();
    expect(buttonTitle).toEqual(title.toUpperCase());
    // the only one case when upperCase !== lowerCase
    if (title !== '') {
      expect(buttonTitle).not.toEqual(title.toLowerCase());
    }
  });
});
