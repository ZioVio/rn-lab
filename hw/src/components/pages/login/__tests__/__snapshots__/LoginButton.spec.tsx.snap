// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="LoginButton.TouchableHighlight"
>
  <View
    accessible={true}
    style={Object {}}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ff6600",
          "borderRadius": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fffffe",
            "fontSize": 30,
            "paddingVertical": 2,
          }
        }
        testID="LoginButton.Text"
      >
        SOME TITLE
      </Text>
    </View>
  </View>
</RNGestureHandlerButton>
`;
